src/app/app.service.ts

```diff
@@ -18,20 +18,30 @@ export class AppConfig {
   private _config
   private isBrowser
 
+  private PORTS_TO_EXCLUDE = ["80", "4000", "8080"];
+
   constructor(@Inject(PLATFORM_ID) private platformId, private injector: Injector) {
     this.isBrowser = isPlatformBrowser(platformId);
+
     // Identify hostname from request or client side
+    this.isBrowser = false;
     if (this.isBrowser) {
-      this.clientHostname = window.location.hostname
-    } else{
+      this.clientHostname = this.getHostName(window.location.hostname, window.location.port);
+    } else {
       let req = this.injector.get('request');
```

Comment:
I refactored this code to use the `getHostname` function because I thought there could be problems if we had a port number like `8001`.

Reply:
I poked around a bit but I am generally confused why we would exclude any ports at all? Maybe something with the server side rendering?

Reply:
Maybe? I didn't want to change the current functionality.

Reply:
It seems like we only filtered out the ports when dealing with the server side rendering? Wondering if we should narrow the scope to just that part of the if/else branch?

Reply:
No, the problem is when `isBrowser` is set to `true`. Before we were only passing `window.location.hostname` back, which was causing the share URL for proxies to not work. So at a minimum we need to pass back the port for the `isBrowser` section of the if/then. I filtered out the three ports so that you would get a consistent URL whether `isBrowser` was true or false.

The share URL actually should be fine most of the time when `isBrowser` is false because it only filters out those three ports. (So a port of 2443, which was the issue in the ticket, would be fine.) I changed that code just because if the port were, say, 8001, then it would return {hostname}01 instead of {hostname}:8001. It's probably an edge case but it seemed worth fixing.

Reply:
and I am assuming there was a specific reason for the original filtering out of 8080, 4000, and 80, but I do not know what that reason was.

Reply:
Gotcha. Ya I just didn't know if there was a reason we were not filtering out ports when `isBrowser` is true or something like that.

Reply:
we were actually filtering out all ports when `isBrowser` was true. But I don't know whether that was intentional or the original coder expected that `window.location.hostname` would return the port the way the `host` in `request` does.